pkg/cloud/instance.go

```diff
@@ -139,35 +139,48 @@ func (c *client) ResolveTemplate(
 // disk offering name matches name provided in spec.
 // If disk offering ID is not provided, the disk offering name is used to retrieve disk offering ID.
 func (c *client) ResolveDiskOffering(csMachine *infrav1.CloudStackMachine) (diskOfferingID string, retErr error) {
-	if len(csMachine.Spec.DiskOffering.ID) > 0 {
-		csDiskOffering, count, err := c.cs.DiskOffering.GetDiskOfferingByID(csMachine.Spec.DiskOffering.ID)
+	diskOfferingID = csMachine.Spec.DiskOffering.ID
```

Comment:
I'm wondering if maybe we should break DiskOfferings into their own file and Interface?

I'm fine either way, but I know there's a lot going on in the Instance interface and file already.